pkg/service/endpoints/gathering_rules.go

```diff
@@ -0,0 +1,42 @@
+package endpoints
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/openshift/insights-operator-conditional-gathering/pkg/service"
+)
+
+type GatheringRulesResponse struct {
+	Version string      `json:"version"`
+	Rules   interface{} `json:"rules"`
+}
+
+func MakeGatheringRulesEndpoint(svc service.Interface) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		rules, err := svc.Rules()
```

Comment:
Is it reading the same file from the disk every time?

Reply:
Right now, yes. Every request reads the file from the disk. Later I'll store it in memory at the `repository.go`.

Reply:
I'd just load them when the program starts and get rid of all these abstractions, we're gonna have the rules in the container image so there's no need to update them during the service runtime, but up to you

Reply:
I'm planning to do it... I think that it will not be a problem to have it in memory 'cause the size isn't that expressive. I'll push the code that does it today.

Reply:
very probably i missed something, but IIRC the service just can return the content of static file, w/o the need to do the magic with JSON encoding and decoding

Reply:
I don't know... I prefer to encode and decode, it will guarantee that the data that we want is being sent in the correct format. Trust that the file will have the expected format, is some price that I don't think is worth paying for. And it also means that line #14 (at this exact same file) needs some work, to not use `interface{}`.

I was also checking the benefits of having it in memory or read every time from disk, and I found none (at least for now)... Maybe it is too small? Maybe my disk is too fast? I don't know... but I can allow it by configuration.

Reply:
ok it sounds legit.

Btw having a cache of data files. We do it in Content service and also Smart Proxy for all rule content (including internal rules) and w/o any problems. So it depends - if your json files are similar in size as all rule content, it's safe to use memory.

Reply:
Well, the format is checked by the schema in CI. Later we may add some filtering and we would need decoding, but who knows.

Reply:
Btw, there's this thing https://pkg.go.dev/embed probably there's no point, but it's interesting.

Reply:
I think we need to store them in memory, because it's easy to implement (just load the file during the program start and store it in a variable), will make it more reliable and a little faster.

Reply:
We gonna have it in memory and also the option to disable it, just in case. So, I guess... no worries.

Reply:
Having an option to turn on/off memory caching sounds like over engineering for me, if we can't load the file with rules during the program start we won't be able to do it later either and they aren't gonna change during program runtime so we don't need to update them.

Reply:
Fine by me.